---
title: "Non-local priors"
author: "Ville Karhunen"
date: "`r format(Sys.Date(), '%d.%m.%Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{nonlocalpriors}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r workinprogress, echo = FALSE, out.width = '20%' }

knitr::include_graphics("work-in-progress.png")

```




```{r setup}

# load required package
library(finimom)

```

Non-local densities are a family of probability distributions with zero density at the null parameter value.
One of the members of this probability distribution family is a product inverse-moment (piMOM) prior for a parameter vector $\boldsymbol{\beta} \in \mathbb{R}^d$, with density

\[
f(\boldsymbol{\beta}|\tau, r) = \prod_{k = 1}^d\frac{\tau^{r/2}}{\Gamma(r/2)}|\beta_k|^{-(r+1)}\exp \left( -\frac{\tau}{\beta_k^2} \right),
\]

with parameters $\tau, r > 0$.
The parameter $\tau$ controls the spread of the distribution away from the null value, and $r$ controls the tail behaviour of the distribution (Johnson, Rossell 2012).
Here, we only consider the case of $r = 1$, which leads to Cauchy-like tails in the distribution.

The choice of $\tau$ can be set such that $\mathbb{P}(|\beta| < q_\beta) = q$.
For genetic analyses, Sanyal et al. (2019) suggested $q_\beta = 0.05$ and $q = 0.01$.
We can obtain the corresponding quantile from the relation of the inverse-moment prior with the (inverse) Gamma distribution (Johnson, Rossell 2010):

```{r invgamma}

betaq <- 0.05
q <- 0.01

qgamma(1 - q, shape = 0.5, scale = 1)*betaq^2

```


The marginal density with $\tau = 0.0083$ is reminiscent of the Gaussian distribution with $\sigma^2 = 0.15$, suggested by Wakefield (2017) and commonly used as the prior for genetic effects (e.g. Giambartolomei et al. 2014):

```{r comparison}

curve(dnorm(x, 0, 0.15), -0.8, 0.8,
      col = "red", lty = 2, 
      xlab = bquote(beta), ylab = "Density")
curve(dimom(x, tau = 0.0083, r = 1), lty = 1, add = T)


```

The key differences between the two distributions are that (i) the non-local prior density approaches 0 when $|\beta| \rightarrow 0$, and (ii) the tails of the non-local density are thicker.
Due to (i), non-local priors apply a strong penalty when any of its components are close to zero, therefore making them suitable for variable selection.
As for (ii), the Cauchy-like tails protect from over-shrinkage of true large effects.


# References

Giambartolomei et al. (2014). Bayesian Test for Colocalisation between Pairs of Genetic Association Studies Using Summary Statistics. *PLoS Genet*.

Johnson, Rossell (2010). On the use of non-local prior densities in Bayesian hypothesis tests. *J. R. Statist. Soc. B*.

Johnson, Rossell (2012). Bayesian Model Selection in High-Dimensional Settings. *J Am Stat Assoc*.

Sanyal et al. (2019). GWASinlps: non-local prior based iterative SNP selection tool for genome-wide association studies. *Bioinformatics*.

Wakefield (2009). Bayes factors for genome-wide association studies: comparison with P-values. *Genet Epidemiol*.

# Session information

```{r sessioninfo}

sessionInfo()

```
