<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Genetic fine-mapping for summarised data using product inverse-moment prior for non-zero effect sizes, and a beta-binomial prior for model dimension.">
<title>Fine-mapping Using Inverse-Moment Prior • finimom</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Fine-mapping Using Inverse-Moment Prior">
<meta property="og:description" content="Genetic fine-mapping for summarised data using product inverse-moment prior for non-zero effect sizes, and a beta-binomial prior for model dimension.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">finimom</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Unreleased version">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/minimal-example.html">Tutorial</a>
    <a class="dropdown-item" href="articles/nonlocalpriors.html">Non-local priors for effect size</a>
    <a class="dropdown-item" href="articles/betabinprior.html">Prior for model dimension</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="finimom-">finimom <img src="reference/figures/finimom.png" align="right" width="125"><a class="anchor" aria-label="anchor" href="#finimom-"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>This R package implements a method for fine-mapping summarised genetic data using product inverse-moment prior for the effect sizes, and a beta-binomial prior for the model dimension.</p>
<p>The method is described in:</p>
<p>Karhunen V, Launonen I, Järvelin MR, Sebert S, Sillanpää MJ. Genetic fine-mapping from summary data using a nonlocal prior improves detection of multiple causal variants. <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btad396" class="external-link">doi:10.1093/bioinformatics/btad396</a>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>finimom</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"vkarhune/finimom"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="minimal-example">Minimal example<a class="anchor" aria-label="anchor" href="#minimal-example"></a>
</h2>
<p>Here is a minimal example on using finimom:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># load package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vkarhune.github.io/finimom">finimom</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># example data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">exampledata</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 9</span></span>
<span><span class="co">#&gt;  $ betahat   : num [1:363] 0.0804 0.0927 -0.0253 -0.0165 -0.0128 ...</span></span>
<span><span class="co">#&gt;  $ sebetahat : num [1:363] 0.0601 0.0595 0.0968 0.028 0.0276 ...</span></span>
<span><span class="co">#&gt;  $ allelefreq: num [1:363] 0.0527 0.0541 0.0194 0.3751 0.4224 ...</span></span>
<span><span class="co">#&gt;  $ truebetas : num [1:363] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  $ causals   : num [1:3] 95 314 329</span></span>
<span><span class="co">#&gt;  $ n         : int 2763</span></span>
<span><span class="co">#&gt;  $ n_ref     : int 2637</span></span>
<span><span class="co">#&gt;  $ insampleLD: num [1:363, 1:363] 1 0.971 0.545 0.288 0.244 ...</span></span>
<span><span class="co">#&gt;  $ refLD     : num [1:363, 1:363] 1 0.972 0.558 0.3 0.268 ...</span></span>
<span></span>
<span><span class="co"># effect size estimates and their standard errors</span></span>
<span><span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">betahat</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">sebetahat</span></span>
<span></span>
<span><span class="co"># allele frequencies</span></span>
<span><span class="va">eaf</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">allelefreq</span></span>
<span></span>
<span><span class="co"># LD matrix</span></span>
<span><span class="va">R</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">insampleLD</span></span>
<span></span>
<span><span class="co"># run finimom</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/finimom.html">finimom</a></span><span class="op">(</span>beta <span class="op">=</span> <span class="va">beta</span>, se <span class="op">=</span> <span class="va">se</span>, eaf <span class="op">=</span> <span class="va">eaf</span>, R <span class="op">=</span> <span class="va">R</span>,</span>
<span>               n <span class="op">=</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">n</span>,</span>
<span>               insampleLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               verbose <span class="op">=</span> <span class="cn">TRUE</span>, ala <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># set ala = TRUE for approximate Laplace</span></span>
<span><span class="co">#&gt; Calculating tau based on the sample size.</span></span>
<span><span class="co">#&gt; Clumping variants at r2=0.98</span></span>
<span><span class="co">#&gt; Sampling from the posterior...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 12500 iterations done in 0.71 seconds</span></span>
<span></span>
<span><span class="co"># output: credible sets</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">sets</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  [1] 295 300 301 309 310 314 315 316 317 322 327 330 342 345 351 353</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt;  [1] 313 320 321 323 325 326 329 331 333 337 339 340 341 343 344 348 350 352 355</span></span>
<span><span class="co">#&gt; [20] 360 362</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;  [1]  39  55  63  68  69  72  73  74  75  76  77  78  79  81  82  83  85  86  87</span></span>
<span><span class="co">#&gt; [20]  88  90  92  93  94  95  96  97  99 114 116 117 118 124 126 128 129 131 133</span></span>
<span><span class="co">#&gt; [39] 136 147 149 150 151 159</span></span>
<span></span>
<span><span class="co"># compare with the true causal variants</span></span>
<span><span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span></span>
<span><span class="co">#&gt; [1]  95 314 329</span></span>
<span></span>
<span><span class="co"># check whether the true causals were captured in the credible sets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] FALSE  TRUE FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; [1]  TRUE FALSE FALSE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="further-details">Further details<a class="anchor" aria-label="anchor" href="#further-details"></a>
</h2>
<p>Further examples are given in the articles.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing finimom</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ville Karhunen <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-6064-1588" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ville Karhunen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
