<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Non-local priors ‚Ä¢ finimom</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Non-local priors">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">finimom</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/multitrait.html">NEW: Multi-trait fine-mapping tutorial</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial.html">Tutorial</a></li>
    <li><a class="dropdown-item" href="../articles/nonlocalpriors.html">Non-local priors for effect size</a></li>
    <li><a class="dropdown-item" href="../articles/betabinprior.html">Prior for model dimension</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Non-local priors</h1>
                        <h4 data-toc-skip class="author">Ville
Karhunen</h4>
            
            <h4 data-toc-skip class="date">07.11.2024</h4>
      

      <div class="d-none name"><code>nonlocalpriors.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># load required package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vkarhune.github.io/finimom">finimom</a></span><span class="op">)</span></span></code></pre></div>
<p>Non-local densities are a family of probability distributions with
zero density at the null parameter value. One of the members of this
probability distribution family is a product inverse-moment (piMOM)
prior for a parameter vector
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ùõÉ</mi><mo>‚àà</mo><msup><mi>‚Ñù</mi><mi>d</mi></msup></mrow><annotation encoding="application/x-tex">\boldsymbol{\beta} \in \mathbb{R}^d</annotation></semantics></math>,
with density</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mo stretchy="false" form="prefix">(</mo><mi>ùõÉ</mi><mrow><mo stretchy="true" form="prefix">|</mo><mi>œÑ</mi><mo>,</mo><mi>r</mi><mo stretchy="false" form="postfix">)</mo><mo>=</mo><munderover><mo>‚àè</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>d</mi></munderover><mfrac><msup><mi>œÑ</mi><mrow><mi>r</mi><mi>/</mi><mn>2</mn></mrow></msup><mrow><mi>Œì</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mi>/</mi><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo stretchy="true" form="postfix">|</mo></mrow><msub><mi>Œ≤</mi><mi>k</mi></msub><msup><mo stretchy="false" form="prefix">|</mo><mrow><mo>‚àí</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo>+</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msup><mo>exp</mo><mrow><mo stretchy="true" form="prefix">(</mo><mo>‚àí</mo><mfrac><mi>œÑ</mi><msubsup><mi>Œ≤</mi><mi>k</mi><mn>2</mn></msubsup></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo></mrow><annotation encoding="application/x-tex">
f(\boldsymbol{\beta}|\tau, r) = \prod_{k = 1}^d\frac{\tau^{r/2}}{\Gamma(r/2)}|\beta_k|^{-(r+1)}\exp \left( -\frac{\tau}{\beta_k^2} \right),
</annotation></semantics></math></p>
<p>with parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>œÑ</mi><mo>,</mo><mi>r</mi><mo>&gt;</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">\tau, r &gt; 0</annotation></semantics></math>.
The parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>œÑ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
controls the spread of the distribution away from the null value, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>
controls the tail behaviour of the distribution (Johnson, Rossell 2012).
Here, we only consider the case of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>r</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">r = 1</annotation></semantics></math>,
which leads to Cauchy-like tails in the distribution.</p>
<p>The choice of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>œÑ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
can be set such that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>‚Ñô</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mo stretchy="true" form="prefix">|</mo><mi>Œ≤</mi><mo stretchy="true" form="postfix">|</mo></mrow><mo>&lt;</mo><msub><mi>q</mi><mi>Œ≤</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mi>q</mi></mrow><annotation encoding="application/x-tex">\mathbb{P}(|\beta| &lt; q_\beta) = q</annotation></semantics></math>.
For genetic analyses, Sanyal et al.¬†(2019) suggested
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mi>Œ≤</mi></msub><mo>=</mo><mn>0.05</mn></mrow><annotation encoding="application/x-tex">q_\beta = 0.05</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mn>0.01</mn></mrow><annotation encoding="application/x-tex">q = 0.01</annotation></semantics></math>.
We can obtain the corresponding quantile from the relation of the
inverse-moment prior with the (inverse) Gamma distribution (Johnson,
Rossell 2010):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">betaq</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">qgamma</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">q</span>, shape <span class="op">=</span> <span class="fl">0.5</span>, scale <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">*</span><span class="va">betaq</span><span class="op">^</span><span class="fl">2</span></span>
<span><span class="co">#&gt; [1] 0.008293621</span></span></code></pre></div>
<p>The marginal density with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>œÑ</mi><mo>=</mo><mn>0.0083</mn></mrow><annotation encoding="application/x-tex">\tau = 0.0083</annotation></semantics></math>
is reminiscent of the Gaussian distribution with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>œÉ</mi><mn>2</mn></msup><mo>=</mo><mn>0.15</mn></mrow><annotation encoding="application/x-tex">\sigma^2 = 0.15</annotation></semantics></math>,
suggested by Wakefield (2017) and commonly used as the prior for genetic
effects (e.g.¬†Giambartolomei et al.¬†2014):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">dnorm</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0</span>, <span class="fl">0.15</span><span class="op">)</span>, <span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>,</span>
<span>      col <span class="op">=</span> <span class="st">"red"</span>, lty <span class="op">=</span> <span class="fl">2</span>, </span>
<span>      xlab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="fu"><a href="../reference/dimom.html">dimom</a></span><span class="op">(</span><span class="va">x</span>, tau <span class="op">=</span> <span class="fl">0.0083</span>, r <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">1</span>, add <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="nonlocalpriors_files/figure-html/comparison-1.png" width="700"></p>
<p>The key differences between the two distributions are that (i) the
non-local prior density approaches 0 when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">|</mo><mi>Œ≤</mi><mo stretchy="true" form="postfix">|</mo></mrow><mo>‚Üí</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">|\beta| \rightarrow 0</annotation></semantics></math>,
and (ii) the tails of the non-local density are thicker. Due to (i),
non-local priors apply a strong penalty when any of its components are
close to zero, therefore making them suitable for variable selection. As
for (ii), the Cauchy-like tails protect from over-shrinkage of true
large effects.</p>
<p>Note that the above example with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>œÑ</mi><mo>=</mo><mn>0.0083</mn></mrow><annotation encoding="application/x-tex">\tau = 0.0083</annotation></semantics></math>
is for illustrative purposes only ‚Äì it is recommended to let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>œÑ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
be estimated based on the data (please see Karhunen et al.¬†(2023) for
details).</p>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Giambartolomei et al.¬†(2014). Bayesian Test for Colocalisation
between Pairs of Genetic Association Studies Using Summary Statistics.
<em>PLoS Genet</em>.</p>
<p>Johnson, Rossell (2010). On the use of non-local prior densities in
Bayesian hypothesis tests. <em>J. R. Statist. Soc. B</em>.</p>
<p>Johnson, Rossell (2012). Bayesian Model Selection in High-Dimensional
Settings. <em>J Am Stat Assoc</em>.</p>
<p>Karhunen et al.¬†(2023). Genetic fine-mapping from summary data using
a nonlocal prior improves the detection of multiple causal variants.
<em>Bioinformatics</em>.</p>
<p>Sanyal et al.¬†(2019). GWASinlps: non-local prior based iterative SNP
selection tool for genome-wide association studies.
<em>Bioinformatics</em>.</p>
<p>Wakefield (2009). Bayes factors for genome-wide association studies:
comparison with P-values. <em>Genet Epidemiol</em>.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] finimom_0.2.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] digest_0.6.37     desc_1.4.3        R6_2.5.1          fastmap_1.2.0    </span></span>
<span><span class="co">#&gt;  [5] xfun_0.49         cachem_1.1.0      knitr_1.48        htmltools_0.5.8.1</span></span>
<span><span class="co">#&gt;  [9] rmarkdown_2.29    lifecycle_1.0.4   cli_3.6.3         sass_0.4.9       </span></span>
<span><span class="co">#&gt; [13] pkgdown_2.1.1     textshaping_0.4.0 jquerylib_0.1.4   systemfonts_1.1.0</span></span>
<span><span class="co">#&gt; [17] compiler_4.4.2    highr_0.11        tools_4.4.2       ragg_1.3.3       </span></span>
<span><span class="co">#&gt; [21] evaluate_1.0.1    bslib_0.8.0       Rcpp_1.0.13-1     yaml_2.3.10      </span></span>
<span><span class="co">#&gt; [25] jsonlite_1.8.9    rlang_1.1.4       fs_1.6.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ville Karhunen.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
