<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="finimom">
<title>Non-local priors • finimom</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Non-local priors">
<meta property="og:description" content="finimom">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">finimom</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/tutorial.html">Tutorial</a>
    <a class="dropdown-item" href="../articles/nonlocalpriors.html">Non-local priors for effect size</a>
    <a class="dropdown-item" href="../articles/betabinprior.html">Prior for model dimension</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Non-local priors</h1>
                        <h4 data-toc-skip class="author">Ville
Karhunen</h4>
            
            <h4 data-toc-skip class="date">27.10.2023</h4>
      
      
      <div class="d-none name"><code>nonlocalpriors.Rmd</code></div>
    </div>

    
    
<p><img src="work-in-progress.png" width="20%"></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># load required package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vkarhune.github.io/finimom">finimom</a></span><span class="op">)</span></span></code></pre></div>
<p>Non-local densities are a family of probability distributions with
zero density at the null parameter value. One of the members of this
probability distribution family is a product inverse-moment (piMOM)
prior for a parameter vector <span class="math inline">\(\boldsymbol{\beta} \in \mathbb{R}^d\)</span>, with
density</p>
<p><span class="math display">\[
f(\boldsymbol{\beta}|\tau, r) = \prod_{k =
1}^d\frac{\tau^{r/2}}{\Gamma(r/2)}|\beta_k|^{-(r+1)}\exp \left(
-\frac{\tau}{\beta_k^2} \right),
\]</span></p>
<p>with parameters <span class="math inline">\(\tau, r &gt; 0\)</span>.
The parameter <span class="math inline">\(\tau\)</span> controls the
spread of the distribution away from the null value, and <span class="math inline">\(r\)</span> controls the tail behaviour of the
distribution (Johnson, Rossell 2012). Here, we only consider the case of
<span class="math inline">\(r = 1\)</span>, which leads to Cauchy-like
tails in the distribution.</p>
<p>The choice of <span class="math inline">\(\tau\)</span> can be set
such that <span class="math inline">\(\mathbb{P}(|\beta| &lt; q_\beta) =
q\)</span>. For genetic analyses, Sanyal et al. (2019) suggested <span class="math inline">\(q_\beta = 0.05\)</span> and <span class="math inline">\(q = 0.01\)</span>. We can obtain the corresponding
quantile from the relation of the inverse-moment prior with the
(inverse) Gamma distribution (Johnson, Rossell 2010):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">betaq</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">qgamma</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">q</span>, shape <span class="op">=</span> <span class="fl">0.5</span>, scale <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">*</span><span class="va">betaq</span><span class="op">^</span><span class="fl">2</span></span>
<span><span class="co">#&gt; [1] 0.008293621</span></span></code></pre></div>
<p>The marginal density with <span class="math inline">\(\tau =
0.0083\)</span> is reminiscent of the Gaussian distribution with <span class="math inline">\(\sigma^2 = 0.15\)</span>, suggested by Wakefield
(2017) and commonly used as the prior for genetic effects
(e.g. Giambartolomei et al. 2014):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">dnorm</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0</span>, <span class="fl">0.15</span><span class="op">)</span>, <span class="op">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>,</span>
<span>      col <span class="op">=</span> <span class="st">"red"</span>, lty <span class="op">=</span> <span class="fl">2</span>, </span>
<span>      xlab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="fu"><a href="../reference/dimom.html">dimom</a></span><span class="op">(</span><span class="va">x</span>, tau <span class="op">=</span> <span class="fl">0.0083</span>, r <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">1</span>, add <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="nonlocalpriors_files/figure-html/comparison-1.png" width="700"></p>
<p>The key differences between the two distributions are that (i) the
non-local prior density approaches 0 when <span class="math inline">\(|\beta| \rightarrow 0\)</span>, and (ii) the tails
of the non-local density are thicker. Due to (i), non-local priors apply
a strong penalty when any of its components are close to zero, therefore
making them suitable for variable selection. As for (ii), the
Cauchy-like tails protect from over-shrinkage of true large effects.</p>
<p>Note that the above example with <span class="math inline">\(\tau =
0.0083\)</span> is for illustrative purposes only – it is recommended to
let <span class="math inline">\(\tau\)</span> be estimated based on the
data (please see Karhunen et al. (2023) for details).</p>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Giambartolomei et al. (2014). Bayesian Test for Colocalisation
between Pairs of Genetic Association Studies Using Summary Statistics.
<em>PLoS Genet</em>.</p>
<p>Johnson, Rossell (2010). On the use of non-local prior densities in
Bayesian hypothesis tests. <em>J. R. Statist. Soc. B</em>.</p>
<p>Johnson, Rossell (2012). Bayesian Model Selection in High-Dimensional
Settings. <em>J Am Stat Assoc</em>.</p>
<p>Karhunen et al. (2023). Genetic fine-mapping from summary data using
a nonlocal prior improves the detection of multiple causal variants.
<em>Bioinformatics</em>.</p>
<p>Sanyal et al. (2019). GWASinlps: non-local prior based iterative SNP
selection tool for genome-wide association studies.
<em>Bioinformatics</em>.</p>
<p>Wakefield (2009). Bayes factors for genome-wide association studies:
comparison with P-values. <em>Genet Epidemiol</em>.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] finimom_0.1.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] vctrs_0.6.4       cli_3.6.1         knitr_1.44        rlang_1.1.1      </span></span>
<span><span class="co">#&gt;  [5] xfun_0.40         stringi_1.7.12    purrr_1.0.2       textshaping_0.3.7</span></span>
<span><span class="co">#&gt;  [9] jsonlite_1.8.7    glue_1.6.2        rprojroot_2.0.3   htmltools_0.5.6.1</span></span>
<span><span class="co">#&gt; [13] ragg_1.2.6        sass_0.4.7        rmarkdown_2.25    evaluate_0.22    </span></span>
<span><span class="co">#&gt; [17] jquerylib_0.1.4   fastmap_1.1.1     yaml_2.3.7        lifecycle_1.0.3  </span></span>
<span><span class="co">#&gt; [21] memoise_2.0.1     stringr_1.5.0     compiler_4.3.1    fs_1.6.3         </span></span>
<span><span class="co">#&gt; [25] Rcpp_1.0.11       systemfonts_1.0.5 digest_0.6.33     R6_2.5.1         </span></span>
<span><span class="co">#&gt; [29] magrittr_2.0.3    bslib_0.5.1       tools_4.3.1       pkgdown_2.0.7    </span></span>
<span><span class="co">#&gt; [33] cachem_1.0.8      desc_1.4.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ville Karhunen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
