<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Tutorial • finimom</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">finimom</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/tutorial.html">Tutorial</a></li>
    <li><a class="dropdown-item" href="../articles/nonlocalpriors.html">Non-local priors for effect size</a></li>
    <li><a class="dropdown-item" href="../articles/betabinprior.html">Prior for model dimension</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Tutorial</h1>
                        <h4 data-toc-skip class="author">Ville
Karhunen</h4>
            
            <h4 data-toc-skip class="date">19.09.2024</h4>
      

      <div class="d-none name"><code>tutorial.Rmd</code></div>
    </div>

    
    
<p><img src="work-in-progress.png" width="20%"></p>
<p>This is a step-by-step tutorial for conducting genetic fine-mapping
on summary data using <em>finimom</em> package.</p>
<div class="section level2">
<h2 id="package-installation-and-load">Package installation and load<a class="anchor" aria-label="anchor" href="#package-installation-and-load"></a>
</h2>
<p>Make sure that you are using the most recent version of the
package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># install package</span></span>
<span><span class="co"># if needed, install also 'remotes' package via 'install.packages("remotes")'</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"vkarhune/finimom"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load required package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vkarhune.github.io/finimom">finimom</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>For the input data, the method requires</p>
<ol style="list-style-type: decimal">
<li><p>The <strong>effect sizes</strong> and their <strong>standard
errors</strong> of variant-phenotype associations within the genetic
region of interest from a genome-wide association study (<em>source
GWAS</em>).</p></li>
<li><p>The <strong>linkage disequilibrium (LD) matrix</strong> of the
variants within the genetic region.</p></li>
</ol>
<p>Additionally, it is recommended to have the <strong>effect allele
frequencies</strong>, <strong>source GWAS sample size</strong>, and
<strong>variance of the phenotype</strong> (in the source GWAS). If the
phenotype variance
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>≠</mo><annotation encoding="application/x-tex">\neq</annotation></semantics></math>
1, then please scale the effect sizes and standard errors to correspond
to standard-deviation units.</p>
<p>If you have only Z-scores available, please see the corresponding
section below.</p>
<div class="section level3">
<h3 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>The R package comes with a simulated example dataset of 363 genetic
variants:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">exampledata</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 9</span></span>
<span><span class="co">#&gt;  $ betahat   : num [1:363] 0.0804 0.0927 -0.0253 -0.0165 -0.0128 ...</span></span>
<span><span class="co">#&gt;  $ sebetahat : num [1:363] 0.0601 0.0595 0.0968 0.028 0.0276 ...</span></span>
<span><span class="co">#&gt;  $ allelefreq: num [1:363] 0.0527 0.0541 0.0194 0.3751 0.4224 ...</span></span>
<span><span class="co">#&gt;  $ truebetas : num [1:363] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  $ causals   : num [1:3] 95 314 329</span></span>
<span><span class="co">#&gt;  $ n         : int 2763</span></span>
<span><span class="co">#&gt;  $ n_ref     : int 2637</span></span>
<span><span class="co">#&gt;  $ insampleLD: num [1:363, 1:363] 1 0.971 0.545 0.288 0.244 ...</span></span>
<span><span class="co">#&gt;  $ refLD     : num [1:363, 1:363] 1 0.972 0.558 0.3 0.268 ...</span></span></code></pre></div>
<p>The object <code>exampledata</code> is a list, which has information
on</p>
<ol style="list-style-type: decimal">
<li><p>GWAS estimated effect sizes (<code>betahat</code>),</p></li>
<li><p>the standard errors (<code>sebetahat</code>),</p></li>
<li><p>allele frequencies (<code>allelefreqs</code>),</p></li>
<li><p>true causal effect sizes (<code>truebetas</code>; this is a
simulated dataset so these are known),</p></li>
<li><p>indices of the true causal variants
(<code>causals</code>),</p></li>
<li><p>GWAS source sample size (<code>n</code>)</p></li>
<li><p>LD reference sample size (<code>n_ref</code>)</p></li>
<li><p>in-sample LD matrix (<code>insampleLD</code>)</p></li>
<li><p>reference, or out-of-sample, LD matrix
(<code>refLD</code>)</p></li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># effect size estimates and their standard errors</span></span>
<span><span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">betahat</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">sebetahat</span></span>
<span></span>
<span><span class="co"># Z-scores</span></span>
<span><span class="va">zscores</span> <span class="op">&lt;-</span> <span class="va">beta</span><span class="op">/</span><span class="va">se</span></span>
<span></span>
<span><span class="co"># p-values</span></span>
<span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fl">2</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">pnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">zscores</span><span class="op">)</span>, lower.tail <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># allele frequencies</span></span>
<span><span class="va">eaf</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">allelefreq</span></span>
<span></span>
<span><span class="co"># LD matrix</span></span>
<span><span class="va">R</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">insampleLD</span></span>
<span></span>
<span><span class="co"># Source GWAS sample size</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">n</span></span></code></pre></div>
<p>The summary statistics can be visualised via a Manhattan plot. In the
following figure, the true causal variants are also highlighted.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"Genetic position"</span>, ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="op">-</span><span class="va">log</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span>, pch <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>, legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Causal"</span>, <span class="st">"Non-causal"</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"black"</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial_files/figure-html/manhattan1-1.png" width="700"></p>
<p>The aim of fine-mapping is to identify the true causal variants
(i.e. those in red in the above figure). In FiniMOM, this problem is
answered by <em>credible sets</em> of causal signals.</p>
</div>
</div>
<div class="section level2">
<h2 id="selecting-hyperparameters">Selecting hyperparameters<a class="anchor" aria-label="anchor" href="#selecting-hyperparameters"></a>
</h2>
<p>Two main hyperparameters need to be selected:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>,
which controls the prior for the detectable effect sizes, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>u</mi><annotation encoding="application/x-tex">u</annotation></semantics></math>,
which controls the prior for the number of causal variants. By default,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
is estimated based on the sample size, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">u = 2</annotation></semantics></math>
for in-sample LD matrix, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mo>=</mo><mn>2.25</mn></mrow><annotation encoding="application/x-tex">u = 2.25</annotation></semantics></math>
for out-of-sample LD matrix. See the manuscript and its supplement for
further details.</p>
</div>
<div class="section level2">
<h2 id="running-finimom">Running finimom<a class="anchor" aria-label="anchor" href="#running-finimom"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finimom.html">finimom</a></span><span class="op">(</span>beta <span class="op">=</span> <span class="va">beta</span>, se <span class="op">=</span> <span class="va">se</span>, eaf <span class="op">=</span> <span class="va">eaf</span>, R <span class="op">=</span> <span class="va">R</span>,</span>
<span>               n <span class="op">=</span> <span class="va">n</span>,</span>
<span>               insampleLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               ala <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculating tau based on the sample size.</span></span>
<span><span class="co">#&gt; Clumping variants at r2=0.98</span></span>
<span><span class="co">#&gt; Sampling from the posterior...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 12500 iterations done in 0.31 seconds</span></span></code></pre></div>
<p>The output is a list, where the first element (<code>samples</code>)
has information on the Markov Chain Monte Carlo samples, the second
element (<code>signals</code>) is the posterior distribution for the
number of signals (or credible sets), and the third element is a list of
the obtained 95% credible sets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "samples" "signals" "sets"</span></span>
<span></span>
<span><span class="co"># posterior distribution for the number of signals</span></span>
<span><span class="va">out</span><span class="op">$</span><span class="va">signals</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      3      4      5 </span></span>
<span><span class="co">#&gt; 0.9341 0.0628 0.0031</span></span>
<span></span>
<span><span class="co"># credible sets</span></span>
<span><span class="va">out</span><span class="op">$</span><span class="va">sets</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  [1] 295 300 301 309 310 314 315 316 317 322 327 330 342 345 351 353</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt;  [1] 313 320 321 323 325 326 329 331 333 337 339 340 341 343 344 348 350 352 355</span></span>
<span><span class="co">#&gt; [20] 360 362</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;  [1]  39  55  63  68  69  72  73  74  75  76  77  78  79  81  82  83  85  86  87</span></span>
<span><span class="co">#&gt; [20]  88  90  92  93  94  95  96  97  99 114 116 117 118 124 126 128 129 131 133</span></span>
<span><span class="co">#&gt; [39] 136 147 149 150 151 159</span></span></code></pre></div>
<p>To obtain the variant-wise posterior inclusion probabilities, run</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">pips</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_pips.html">get_pips</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualising-results">Visualising results<a class="anchor" aria-label="anchor" href="#visualising-results"></a>
</h2>
<p>One can extract the credible sets and visualise results e.g. as
follows:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"Genetic position"</span>, ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/bquote.html" class="external-link">bquote</a></span><span class="op">(</span><span class="op">-</span><span class="va">log</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">exampledata</span><span class="op">$</span><span class="va">causals</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span>, pch <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"dodgerblue"</span>, pch <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"orange"</span>, pch <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">out</span><span class="op">$</span><span class="va">sets</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"darkolivegreen"</span>, pch <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>, legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Causal"</span>, <span class="st">"Non-causal"</span>, <span class="st">"Credible set 1"</span>, <span class="st">"Credible set 2"</span>, <span class="st">"Credible set 3"</span><span class="op">)</span>,</span>
<span> col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"black"</span>, <span class="st">"dodgerblue"</span>, <span class="st">"orange"</span>, <span class="st">"darkolivegreen"</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">16</span>, <span class="fl">16</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial_files/figure-html/visualise-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="using-out-of-sample-ld-matrix">Using out-of-sample LD matrix<a class="anchor" aria-label="anchor" href="#using-out-of-sample-ld-matrix"></a>
</h2>
<p>If using LD matrix from a reference panel, set
<code>insampleLD = FALSE</code>. In addition, it is recommended to check
for any LD discrepancies by <code>check_ld = TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">R_ref</span> <span class="op">&lt;-</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">refLD</span></span>
<span></span>
<span><span class="va">out_refld</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finimom.html">finimom</a></span><span class="op">(</span>beta <span class="op">=</span> <span class="va">beta</span>, se <span class="op">=</span> <span class="va">se</span>, eaf <span class="op">=</span> <span class="va">eaf</span>, R <span class="op">=</span> <span class="va">R_ref</span>,</span>
<span>               n <span class="op">=</span> <span class="va">n</span>,</span>
<span>               insampleLD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>               check_ld <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               ala <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculating tau based on the sample size.</span></span>
<span><span class="co">#&gt; Clumping variants at r2=0.98</span></span>
<span><span class="co">#&gt; Sampling from the posterior...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 12500 iterations done in 0.37 seconds</span></span>
<span></span>
<span><span class="va">out_refld</span><span class="op">$</span><span class="va">sets</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  [1] 124 126 128 129 133 294 295 296 300 301 309 310 314 315 316 317 322 327 330</span></span>
<span><span class="co">#&gt; [20] 342 345 351 353</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt;  [1] 313 320 321 323 325 326 329 331 333 337 339 340 341 343 344 348 350 352 355</span></span>
<span><span class="co">#&gt; [20] 360 362</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;  [1]  39  55  56  57  68  69  72  73  74  75  76  77  78  79  81  82  83  85  86</span></span>
<span><span class="co">#&gt; [20]  87  88  90  92  93  94  95  96  97  99 105 108 109 111 112 113 114 116 117</span></span>
<span><span class="co">#&gt; [39] 118 124 126 128 129 131 133 136 147 150 151 159 345</span></span></code></pre></div>
<p>In this case, some variants are in multiple credible sets. This
behaviour can be mitigated by selecting a desired purity level for the
credible sets (minimum absolute correlation for the variants within a
credible set) by using option <code>purity</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">out_refld2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finimom.html">finimom</a></span><span class="op">(</span>beta <span class="op">=</span> <span class="va">beta</span>, se <span class="op">=</span> <span class="va">se</span>, eaf <span class="op">=</span> <span class="va">eaf</span>, R <span class="op">=</span> <span class="va">R_ref</span>,</span>
<span>               n <span class="op">=</span> <span class="va">n</span>,</span>
<span>               insampleLD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>               check_ld <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               ala <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               purity <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculating tau based on the sample size.</span></span>
<span><span class="co">#&gt; Clumping variants at r2=0.98</span></span>
<span><span class="co">#&gt; Sampling from the posterior...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 12500 iterations done in 0.30 seconds</span></span>
<span></span>
<span><span class="va">out_refld2</span><span class="op">$</span><span class="va">sets</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  [1] 294 295 296 300 301 309 310 314 315 316 317 322 327 330 342 345 351 353</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt;  [1] 313 320 321 323 325 326 329 331 333 337 339 340 341 343 344 348 350 352 355</span></span>
<span><span class="co">#&gt; [20] 360 362</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;  [1]  39  55  57  68  69  72  73  74  75  76  77  78  79  81  82  83  85  86  87</span></span>
<span><span class="co">#&gt; [20]  88  90  92  93  94  95  96  97  99 114 124 126 128 129 131 133 147 150 151</span></span>
<span><span class="co">#&gt; [39] 159</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="using-z-scores">Using Z scores<a class="anchor" aria-label="anchor" href="#using-z-scores"></a>
</h2>
<p>If only Z-scores are available (i.e. no information on the effect
sizes and their standard errors), then for a standardized (mean = 0,
standard deviation = 1) genetic variant and phenotype,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">SE</mtext><mrow><mo stretchy="true" form="prefix">(</mo><mover><mi>β</mi><mo accent="true">̂</mo></mover><mo stretchy="true" form="postfix">)</mo></mrow><mo>≈</mo><mn>1</mn><mi>/</mi><msqrt><mi>n</mi></msqrt></mrow><annotation encoding="application/x-tex">\text{SE}(\hat{\beta}) \approx 1/\sqrt{n}</annotation></semantics></math>.
As these are (approximately) on the standardized scale, use option
<code>standardize = FALSE</code> for avoiding the standardization that
is done by default.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span></span>
<span><span class="va">approx_se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">zscores</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">approx_beta</span> <span class="op">&lt;-</span> <span class="va">zscores</span><span class="op">*</span><span class="va">approx_se</span></span>
<span></span>
<span><span class="va">out_zscore_only</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finimom.html">finimom</a></span><span class="op">(</span>beta <span class="op">=</span> <span class="va">approx_beta</span>, se <span class="op">=</span> <span class="va">approx_se</span>,</span>
<span>                           eaf <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                           R <span class="op">=</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">insampleLD</span>,</span>
<span>                           n <span class="op">=</span> <span class="va">exampledata</span><span class="op">$</span><span class="va">n</span>,</span>
<span>                           standardize <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># &lt;--- use this option for Z-scores</span></span>
<span>                           insampleLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                           ala <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculating tau based on the sample size.</span></span>
<span><span class="co">#&gt; Clumping variants at r2=0.98</span></span>
<span><span class="co">#&gt; Sampling from the posterior...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 12500 iterations done in 0.30 seconds</span></span>
<span></span>
<span><span class="va">out_zscore_only</span><span class="op">$</span><span class="va">sets</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  [1] 295 300 309 310 314 315 317 322 327 330 342 345 351 353</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt;  [1] 313 320 321 323 325 326 329 331 333 337 339 340 341 343 344 348 350 352 355</span></span>
<span><span class="co">#&gt; [20] 360 362</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;  [1]  55  63  68  69  72  73  74  75  76  77  78  79  81  82  83  85  86  87  88</span></span>
<span><span class="co">#&gt; [20]  90  92  93  94  95  96  97  99 114 116 117 118 124 126 128 129 131 133 136</span></span>
<span><span class="co">#&gt; [39] 147 150 151 159</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] finimom_0.1.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] digest_0.6.37     desc_1.4.3        R6_2.5.1          fastmap_1.2.0    </span></span>
<span><span class="co">#&gt;  [5] xfun_0.47         cachem_1.1.0      knitr_1.48        htmltools_0.5.8.1</span></span>
<span><span class="co">#&gt;  [9] rmarkdown_2.28    lifecycle_1.0.4   cli_3.6.3         sass_0.4.9       </span></span>
<span><span class="co">#&gt; [13] pkgdown_2.1.1     textshaping_0.4.0 jquerylib_0.1.4   systemfonts_1.1.0</span></span>
<span><span class="co">#&gt; [17] compiler_4.4.1    highr_0.11        tools_4.4.1       ragg_1.3.3       </span></span>
<span><span class="co">#&gt; [21] evaluate_1.0.0    bslib_0.8.0       Rcpp_1.0.13       yaml_2.3.10      </span></span>
<span><span class="co">#&gt; [25] jsonlite_1.8.8    rlang_1.1.4       fs_1.6.4</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ville Karhunen.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
